src/charm.py

```diff
@@ -1,206 +1,176 @@
+                        "-enable-ui-service=true "
+                        "-controller-threads=10 "
+                        "-resync-interval=30 "
+                        "-enable-batch-scheduler=false "
+                        "-enable-metrics=true "
+                        "-metrics-labels=app_type "
+                        f"-metrics-port={self.model.config['metrics-port']} "
+                        "-metrics-endpoint=/metrics "
+                        "-enable-resource-quota-enforcement=false "
+                        "-enable-webhook=true "
+                        f"-webhook-svc-namespace={self.model.name} "
+                        f"-webhook-port={self.model.config['webhook-port']} "
+                        f"-webhook-svc-name={self.model.app.name} "
+                        f"-webhook-config-name={self._mutating_webhook_name} "
+                        f"-webhook-namespace-selector=model.juju.is/name={self.model.name} "
+                        "-webhook-fail-on-error=true"
+                    ),
                 }
             },
+        }
+        return Layer(pebble_layer)
+
+    def _update_layer(self) -> None:
+        """Updates the Pebble configuration layer if changed."""
+
+        current_layer = self.container.get_plan()
+        new_layer = self._spark_operator_layer
+
+        if current_layer.services != new_layer.services:
+            self.container.add_layer(self._container_name, new_layer, combine=True)
+            try:
+                log.info("Pebble plan updated with new configuration, replanning")
+                self.container.replan()
+            except ChangeError as e:
+                log.error(traceback.format_exc())
+                self.unit.status = BlockedStatus("Failed to replan")
+                raise e
+                return
+
+    def _update_webhook_certs(self) -> None:
+        """Push keys and certs files into spark container"""
+
+        self.container.push("/etc/webhook-certs/ca-cert.pem", self._stored.ca, make_dirs=True)
+        self.container.push(
+            "/etc/webhook-certs/server-cert.pem", self._stored.cert, make_dirs=True
         )
-        self.model.unit.status = ActiveStatus()
+        self.container.push("/etc/webhook-certs/server-key.pem", self._stored.key, make_dirs=True)
+        log.info("Pushed webhook keys and certs to spark container")
+
+    def _update_spark_container(self) -> None:
+        if not self.container.can_connect():
+            self.unit.status = MaintenanceStatus("Waiting to connect to spark container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring Spark Charm")
+
+        self._update_webhook_certs()
+        self._update_layer()
+
+        self.unit.status = ActiveStatus()
+
+    def _on_install(self, _):
+        """Event Handler for install event."""
+        self.resource_handler.apply()
+
+        if self.container.can_connect():
+            self._update_webhook_certs()
```

Comment:
Is there a reason for updating webhook certs on install if it's also done on pebble_ready event? In both cases it runs only if the container is ready.

Reply:
In some [previous runs of integration tests on CI](https://github.com/canonical/spark-operator/runs/7524206470?check_suite_focus=true#step:4:91), the spark operator would fail to start because it couldn't find the files. Having these two lines seems to help with that.